### PR TITLE
Migrate interview to sapphire

### PR DESCRIPTION
**Pull request by alexz429**
Migrated interview commands to sapphire

Note that argument validation and parsing is a feature not implemented in sapphire. Will have to rewrite this functionality in the future.

For now, it is *expected* that invalid arguments cause code errors (although the bot will continue operating for new commands)
image
image

### CODE REVIEW

**Review by marko-polo-cheno — APPROVED** [review]
hmm should re delete old code or keep somewhere else for a bit?

File: src/commands/interviewers/interviewer.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  resumeProfile,
+  toggleDomain,
+  upsertInterviewer
+} from '../../components/interview';
+import { BOT_PREFIX } from '../../bot';
+import { getEmojiByName } from '../../components/emojis';
+import _ from 'lodash';
+
+// Extend `SubCommandPluginCommand` instead of `Command`
+export class UserCommand extends SubCommandPluginCommand {
```
Comment (marko-polo-cheno) [inline src/commands/interviewers/interviewer.ts:25]: hold up command naming?

**Review by marko-polo-cheno — CHANGES_REQUESTED** [review]

File: src/commands/interviewers/interviewer.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    const user = await this.container.client.users.fetch(interviewer['user_id']);
+    const userDomainsAddIn = await getInterviewerDomainsString(interviewer['user_id']);
+    if (userDomainsAddIn === '') {
+      return `${user} | [Calendar](${interviewer['link']})\n\n`;
+    } else {
+      return `${user} | [Calendar](${interviewer['link']}) | ${userDomainsAddIn}\n\n`;
+    }
+  }
+
+  RESULTS_PER_PAGE = 6;
```
Comment (marko-polo-cheno) [inline src/commands/interviewers/interviewer.ts:131]: would this live better somewhere else?

**Review by marko-polo-cheno — CHANGES_REQUESTED** [review]
also if screenshots can be provided, plz do

File: src/bot.ts [hunk 3]
```diff
@@ -33,6 +33,8 @@ export const client = new SapphireClient({
   ]
 });
 
+client.fetchPrefix = (): string => BOT_PREFIX;
```
Comment (marko-polo-cheno) [inline src/bot.ts:36]: whats this for?

Comment (alexz429) [reply src/bot.ts:36]: whoops, must've slipped through when I was doing sapphire tests

File: src/commands/interviewers/interviewer.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+import _ from 'lodash';
+import { container } from '@sapphire/pieces';
+import { ApplyOptions } from '@sapphire/decorators';
+
+const RESULTS_PER_PAGE = 6;
+
+@ApplyOptions<SubCommandPluginCommandOptions>({
+  aliases: ['interviewers'],
+  description: 'Handles interviewer functions',
+  detailedDescription: `**Examples:**\n\`${container.client.options.defaultPrefix}interviewer frontend\``,
```
Comment (marko-polo-cheno) [inline src/commands/interviewers/interviewer.ts:32]: import { BOT_PREFIX } from '../../bot';
{ BOT_PREFIX }
shorter + consistent with others

**Review by marko-polo-cheno — CHANGES_REQUESTED** [review]

**Review by marko-polo-cheno — APPROVED** [review]

**Pull request merged**